proteuslib/property_models/tests/property_test_harness.py

```diff
@@ -0,0 +1,532 @@
+                                                         obj_units=obj_units,
+                                                         obj_units_1=pyunits.get_units(obj)))
+
+    @staticmethod
+    def check_block_obj_coverage(blk, obj_dict,
+                         type_tpl=(Param, Var, Expression, Constraint)):
+        """Method to check that all pyomo objects on block are in the testing dictionary.
+
+        Args:
+            blk: pyomo block that will be checked
+            obj_dict: dictionary with pyomo objects and their expected type and units.
+                keys = string name of pyomo object, values = (pyomo type, pyomo units)
+            type_tpl: tuple including all pyomo types that will be assessed
+
+        Returns:
+            None
+            """
+        for obj in blk.component_objects(type_tpl, descend_into=False):
+            obj_str = obj.local_name
+            if obj_str not in obj_dict:
+                raise Exception(
+                    "{obj_str} is on {blk}, but is not included in the testing dictionary. "
+                    "Add {obj_str} to the appropriate dictionary in the configure function"
+                    "that sets up the PropertyUnitTestHarness".format(obj_str=obj_str, blk=blk))
+
+    @pytest.fixture(scope='class')
+    def frame(self):
+        m = ConcreteModel()
+        self.configure_class(m)
+
+        m.fs = FlowsheetBlock(default={"dynamic": False})
+        m.fs.properties = self.prop_pack()
+        m.fs.stream = m.fs.properties.build_state_block(
+            [0], default=self.param_args)
+
+        for (v_str, ind), sf in self.set_default_scaling_dict.items():
+            m.fs.properties.set_default_scaling(v_str, sf, index=ind)
+
+        return m
+
+    @pytest.mark.unit
+    def test_components_phases(self, frame):
+        m = frame
+        # components
+        assert isinstance(m.fs.properties.component_list, Set)
+        assert len(m.fs.properties.component_list) == len(m.component_dict)
+        for c_str in m.component_dict:
+            assert c_str in m.fs.properties.component_list
+            c = getattr(m.fs.properties, c_str)
+            c_type = m.component_dict[c_str]
+            assert isinstance(c, c_type)
+
+        # phases
+        assert isinstance(m.fs.properties.phase_list, Set)
+        assert len(m.fs.properties.phase_list) == len(m.phase_dict)
+        for p_str in m.phase_dict:
+            assert p_str in m.fs.properties.phase_list
+            p = getattr(m.fs.properties, p_str)
+            p_type = m.phase_dict[p_str]
+            assert isinstance(p, p_type)
+
+    @pytest.mark.unit
+    def test_parameters(self, frame):
+        m = frame
+
+        self.check_block_obj(m.fs.properties, m.param_obj_dict)
+        self.check_block_obj_coverage(m.fs.properties, m.param_obj_dict)
+
+        # test that the parameter variables are fixed
+        for v in m.fs.properties.component_objects(Var):
```

Comment:
This presumes that all parameters will be Vars. Do we also need to allow for Params in some cases?

Reply:
There can be Params on the property block, this test just makes sure that any of the parameters that are Vars are fixed.